pkg/substitution/substitution.go

```diff
@@ -234,7 +234,8 @@ func extractExpressionFromString(s, prefix string) (string, bool) {
 	return match[0], true
 }
 
-func extractVariablesFromString(s, prefix string) ([]string, bool, string) {
+// ExtractVariablesFromString extracts variables from an input string s with the given prefix via regex matching.
+func ExtractVariablesFromString(s, prefix string) ([]string, bool, string) {
```

Comment:
can we avoid exporting this function and instead make use of one of the already exported functions in this package? If we have to export it, can you please document the return values, and what happens if the string does not have the given prefix?

Reply:
Maybe an alternative option would be to write a function that gets all the names of workspaces used in a task spec?

Reply:
I might have to write a dedicated function e.g.`ExtractWorkspaceFromString` that calls this function internally. 

The only exported functions in this package are `Validate...` and `Apply...` which I cannot use here. I could move `FindWorkspaceSubstitutionLocationsInTask` in here instead of in `pkg/workspaces` and combine the usage to a single function e.g. `FindWorkspacesRequiredbyTask`.

Reply:
However, `substitution` seems like a more general purpose package than `workspace` so moving workspace specific changes to substitution does not make sense.

Reply:
For now, I have left `ExtractVariablesFromString ` as is since it needs to be called by `pkg/workspaces` which now contains `FindWorkspacesUsedByTask`. I have documented the return values and what happens if the string does not have the given prefix.

Reply:
Thanks! One last comment; I think the boolean variable is redundant with the string slice that's returned (i.e. the bool is just len(stringslice) > 0 )

Reply:
I agree. I can create an issue on it and clean it up in a small PR.